app/src/main/java/com/example/android/weatherapp/viewmodel/HomeScreenViewModel.kt

```diff
@@ -2,35 +2,96 @@ package com.example.android.weatherapp.viewmodel
 import kotlinx.coroutines.launch
 import retrofit2.HttpException
 
-private const val APP_ID= "4dfdfc7144138f43bfa36b5b3a4b097f"
-
 class HomeScreenViewModel : BaseViewModel() {
+
     private val viewModelJob = Job ()
     private val coroutineScope = CoroutineScope(Dispatchers.Main + viewModelJob)
 
-    private val _currentWeather = MutableLiveData<CurrentWeatherModel>()
-    val currentWeather: LiveData<CurrentWeatherModel>
-        get() = _currentWeather
+    val currentWeather = mutableListOf<CurrentWeatherModel>()
+    val threeHourlyWeatherForecast = mutableListOf<ThreeHourlyWeatherModel>()
+    val weeklyWeatherForecast = mutableListOf<WeeklyWeatherModel>()
+
+    private val _eventDataUpdated = MutableLiveData<Boolean>()
+    val eventDataUpdated: LiveData<Boolean>
+        get() = _eventDataUpdated
 
     init {
+      fetchData()
+    }
+
+    private fun fetchData() {
         getCurrentWeather()
+        getForecast()
+    }
+
+    private fun getForecast() {
+        coroutineScope.launch {
+            val forecastObj = WeatherApi.retrofitService.getFiveDaysWeather("Lahore",
+                Helper.APP_ID)
+            try {
+                if(forecastObj.isSuccessful) {
+                    prepareThreeHourlyForecastData(forecastObj.body())
+                    prepareWeeklyForecastData(forecastObj.body())
+                }
+                onDataUpdated()
+            } catch (e: HttpException) {
+                println("Exception ${e.message}")
+            } catch (e: Throwable) {
+                println("Ooops: Something else went wrong")
+            }
+        }
+    }
+
+    private fun prepareThreeHourlyForecastData(forecastObj: ForecastWeatherModel?) {
+        forecastObj?.let {
+            val tempList = forecastObj.list.subList(0,5)
+            for( obj in tempList){
+                threeHourlyWeatherForecast.add(ThreeHourlyWeatherModel(
+                        Helper.getTimeFromDate(obj.dt_txt),
+                        Helper.convertToCelsius(obj.main.temp).toString()
+                    )
+                )
+            }
+        }
+    }
+
+    private fun prepareWeeklyForecastData(forecastObj: ForecastWeatherModel?) {
+        forecastObj?.let {
+            for (obj in 11 until 40 step 8) {
+                val tempObj = forecastObj.list[obj]
+                weeklyWeatherForecast.add(WeeklyWeatherModel(
+                        Helper.getDayFromDate(tempObj.dt_txt),
+                        Helper.formatWeeklyForecastTemperature(
+                            tempObj.main.temp_min,
+                            tempObj.main.temp_max)
+                    )
+                )
+            }
+        }
     }
 
     private fun getCurrentWeather() {
         coroutineScope.launch {
-            var weatherObj = WeatherApi.retrofitService.getCurrentWeather("London", APP_ID)
+            val weatherObj = WeatherApi.retrofitService.getCurrentWeather("Lahore",
+                Helper.APP_ID)
```

Comment:
constants should be in separate file or strings.xml

Reply:
Moved to separate constant file